tests/core/test_document.py

```diff
@@ -0,0 +1,23 @@
+import docx
+
+from ragna.core import DocxDocumentHandler, LocalDocument
+
+
+def get_docx_document(tmp_path, docx_text):
+    document = docx.Document()
+    document.add_heading(docx_text)
+    document.add_paragraph(docx_text)
+    path = tmp_path / "test_document.docx"
+    document.save(path)
+    return LocalDocument.from_path(path)
+
+
+def test_docx(tmp_path):
+    docx_text = "ragna is neat!"
+    tmp_docx_document = get_docx_document(tmp_path, docx_text)
+    assert isinstance(tmp_docx_document.handler, DocxDocumentHandler)
+    pages = list(tmp_docx_document.extract_pages())
+    assert len(pages) == 2
```

Comment:
Why has this document two pages if we only have three words in it?

Reply:
That's a good question. So the document we wrote here has two elements, a heading (different from a header -- it's also considered a paragraph in python_docx, but with special styling on it) and a paragraph, both with the same text in them. The way I've done the Docx handler, it retrieves a different `Page` for each paragraph in the document. We could combine them all beforehand and return a single page, more similar to the .txt handler. 

Current way:
```python
        for paragraph in document_docx.paragraphs:
            text = paragraph.text
            if len(text) > 0:
                yield Page(text=text)
```

Potential way with only one Page:
```python
        text = "\n".join(paragraph.text for paragraph in document_docx.paragraphs)
        yield Page(text=text)
```

Personally, I don't have a huge preference either way.

Reply:
With your insight into the codebase, do you think fewer, bigger pages going to be more performant, or many smaller pages?

Reply:
I just added an assertion for both paragraphs.